src/category_theory/limits/shapes/cartesian_closed.lean

```diff
@@ -33,14 +53,18 @@ This isn't an instance because it's not usually how we want to construct exponen
 def binary_product_exponentiable {C : Type u} [category.{v} C] [has_finite_products.{v} C] {X Y : C}
   (hX : exponentiable X) (hY : exponentiable Y) : exponentiable (X ⨯ Y) :=
 { is_adj :=
-  by haveI := hX.is_adj; haveI := hY.is_adj;
-     exact adjunction.left_adjoint_of_nat_iso (prod_functor_left_comp _ _).symm }
-
-/-- A category `C` is cartesian closed if every object is exponentiable, and it has finite products. -/
-class is_cartesian_closed (C : Type u) [category.{v} C] [has_finite_products.{v} C] :=
-(cart_closed : Π (X : C), exponentiable X)
+  begin
+    haveI := hX.is_adj,
+    haveI := hY.is_adj,
+    exact adjunction.left_adjoint_of_nat_iso (monoidal_category.tensor_left_tensor _ _).symm
+  end }
 
-attribute [instance, priority 100] is_cartesian_closed.cart_closed
+/--
+A category `C` is cartesian closed if it has finite products and every object is exponentiable.
+We define this as `is_closed` with respect to the cartesian monoidal structure.
+-/
+abbreviation is_cartesian_closed (C : Type u) [category.{v} C] [has_finite_products.{v} C] :=
```

Comment:
By the way I called this `is_cartesian_closed` - I'm not sure if it's the best name though, since it doesn't really follow the `is_` pattern throughout the rest of category theory in mathlib - it's not subsingleton for instance. Do you have thoughts on this?

Reply:
Oh --- yes, I guess the pattern would be to drop the `is_`.